src/configs/sort.ts

```diff
@@ -89,6 +89,93 @@ export async function sortPackageJson(): Promise<FlatConfigItem[]> {
             ],
             pathPattern: '^exports.*$',
           },
+          {
+            order: [
+              'prepare',
+              'prepack',
+              'pack',
+              'postpack',
+              'prerelease',
+              'release',
+              'postrelease',
+              'prepublish',
+              'prepublishOnly',
+              'publish',
+              'postpublish',
+              'preinstall',
+              'install',
+              'postinstall',
+              'preuninstall',
+              'uninstall',
+              'postuninstall',
+              'preversion',
+              'version',
+              'postversion',
+              { keyPattern: '^start' },
+              { keyPattern: '^restart' },
+              { keyPattern: '^stop' },
+              'dev',
+              { keyPattern: '^build' },
+              'serve',
+              'preview',
+              'typecheck',
+              'type-check',
+              { keyPattern: '^lint' },
+              { keyPattern: '^test' },
+            ],
+            pathPattern: '^scripts$',
+          },
+          {
+            order: [
+              // client hooks only
+              'pre-commit',
+              'prepare-commit-msg',
+              'commit-msg',
+              'post-commit',
+              'pre-rebase',
+              'post-rewrite',
+              'post-checkout',
+              'post-merge',
+              'pre-push',
+              'pre-auto-gc',
+            ],
+            pathPattern: '^(?:gitHooks|husky|simple-git-hooks)$',
+          },
+          {
+            order: [
+              'root',
+              'env',
+              'globals',
+              'settings',
+              'parser',
+              'parserOptions',
+              'extends',
+              'plugins',
+              'processor',
+              'noInlineConfig',
+              'ignorePatterns',
+              'rules',
+              'overrides',
+            ],
+            pathPattern: '^eslintConfig$',
+          },
+          {
+            order: [
+              'extends',
+              'plugins',
+              'processors',
+              'defaultSeverity',
+              'ignoreFiles',
+              'ignoreDisables',
+              'customSyntax',
+              'reportNeedlessDisables',
+              'reportInvalidScopeDisables',
+              'reportDescriptionlessDisables',
+              'rules',
+              'overrides',
+            ],
+            pathPattern: '^stylelint$',
```

Comment:
For eslint and stylelint, I think it should be recommended to put them in separate configs. I don't think we should support them.